api/topo/topo.proto

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2019-present Open Networking Foundation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+syntax = "proto3";
+
+// Package topo defines interfaces for managing network top
+package topo;
+
+import "google/protobuf/any.proto";
+import "gogoproto/gogo.proto";
+
+// EntityService provides an API for managing entities.
+service topo {
+  // Update one or more entities to the topology
+  rpc Write (WriteRequest) returns (WriteResponse) {
+  }
+
+  // Read one or more entities from topology
+  rpc Read (ReadRequest) returns (ReadResponse) {
+  }
+
+  // Represents the bidirectional stream between
+  // onos-topo and a client for the purpose of
+  // - streaming notifications
+  rpc StreamChannel (stream StreamMessageRequest) returns (stream StreamMessageResponse) {
+  }
+}
+
+message WriteRequest {
+  // The write batch, comprising a list of Update operations
+  repeated Update updates = 1;
+}
+
+message WriteResponse {
+}
+
+message ReadRequest {
+  repeated Object objects = 1;
```

Comment:
should this be just an identifier, as you probably won't know the object until you read it?